pkg/azure/utils.go

```diff
@@ -1133,3 +1133,18 @@ func verifyAzureTags(tags map[string]*string, clusterNameTag, nodeRoleTag string
 
 	return true
 }
+
+// inspectResourceGroup inspects if a given resource group by name is available on Azure.
+// The function will return an error which will be nil if the resource group exists.
+// In case the resource group does not exists on Azure an NotFound error will be returned.
+// If the Azure api call not succeed for other reasons an Internal error will be returned.
+func inspectResourceGroup(ctx context.Context, clients spi.AzureDriverClientsInterface, resourceGroupName string) error {
+	if _, err := clients.GetGroup().Get(ctx, resourceGroupName); err != nil {
+		if NotFound(err) {
+			return status.Error(codes.NotFound, err.Error())
+		}
```

Comment:
If these are MCM codes then the `NotFound` error code is not appropriate for `ListMachine` [ref](https://github.com/gardener/machine-controller-manager/blob/master/docs/development/machine_error_codes.md#listmachines-errors)

Because of that I am not in favor of setting external MCM error codes in utility funcs.

Reply:
Thanks for this. What would you suggest in this case? Simply sending an ok?

Reply:
@himanshu-kun can probably have a more informed opinion about the best reply.

Technically, no RG means that the list of machines is empty, so an ok should be fine. However, as it is right now it will mess with the DeleteMachine response replying with OK instead of NotFound (although an OK and NotFound for DeleteMachine are equivalent).

I would change `inspectResourceGroup` to 

```
resourceGroupExists(...) (bool, error)
```
That way you can discern if RG exists and then you can in ListMachines you can reply with an Ok and empty machine list and for Delete with NotFound

Reply:
yes I agree with @kon-angelo . His proposal looks good to me.

Reply:
I adressed the feedback, but I didn't go with a dedicated function to determine the rg availability.
As we need to handle the situation differently in `DeleteMachine()` and `ListMachines()` it doesn't bring too much benefit and it's not much dry.

In `ListMachines()` I simply return now an empty list of machine so that deletion can be completed. As we have no proper error code for this situation I added a log message for this case.

PTAL :)